ev3dev2/motor.py

```diff
@@ -1924,6 +2037,181 @@ def get_speed_steering(self, steering, speed):
+            abort = True
+
+        if not distance_mm:
+            log.warning("{}: distance_mm is invalid ({}), tank will not move".format(self, distance_mm))
+            abort = True
+
+        if abort:
+            self._set_brake(brake)
+            return
+
+        rotations = distance_mm / self.wheel.circumference_mm
+        log.debug("%s: on_for_rotations distance_mm %s, rotations %s, speed %s" % (self, distance_mm, rotations, speed))
+
+        MoveTank.on_for_rotations(self, speed, speed, rotations, brake, block)
+
+    def _on_arc(self, speed, radius_mm, distance_mm, brake, block, arc_right):
+        """
+        Drive in a circle with 'radius' for 'distance'
+        """
+        abort = False
+
+        if not speed:
+            log.warning("{}: speed is invalid ({}), tank will not move".format(self, speed))
+            abort = True
+
+        if radius_mm <= 0:
+            log.warning("{}: radius_mm is invalid ({}), tank will not move".format(self, radius_mm))
+            abort = True
+
+        if radius_mm < self.min_circle_radius_mm:
+            log.warning("{}: radius_mm is {}, the smallest radius we can make is {}".format(
+                self, radius_mm, self.min_circle_radius_mm))
+            abort = True
+
+        if distance_mm is not None and distance_mm < 0:
+            log.warning("{}: distance_mm is {}, it must be None or greater than 0".format(self, distance_mm))
+            abort = True
+
+        if abort:
+            self._set_brake(brake)
+            return
+
+        # The circle formed at the halfway point between the two wheels is the
+        # circle that must have a radius of radius_mm
+        circle_outer_mm = 2 * pi * (radius_mm + (self.wheel_distance_mm / 2))
+        circle_middle_mm = 2 * pi * radius_mm
+        circle_inner_mm = 2 * pi * (radius_mm - (self.wheel_distance_mm / 2))
+
+        if arc_right:
+            # The left wheel is making the larger circle and will move at 'speed'
+            # The right wheel is making a smaller circle so its speed will be a fraction of the left motor's speed
+            left_speed = speed
+            right_speed = float(circle_inner_mm/circle_outer_mm) * left_speed
+
+        else:
+            # The right wheel is making the larger circle and will move at 'speed'
+            # The left wheel is making a smaller circle so its speed will be a fraction of the right motor's speed
+            right_speed = speed
+            left_speed = float(circle_inner_mm/circle_outer_mm) * right_speed
+
+        log.debug("%s: arc %s, radius %s, distance %s, left-speed %s, right-speed %s, circle_outer_mm %s, circle_middle_mm %s, circle_inner_mm %s" %
+            (self, "right" if arc_right else "left",
+             radius_mm, distance_mm, left_speed, right_speed,
+             circle_outer_mm, circle_middle_mm, circle_inner_mm
+            )
+        )
+
+        if distance_mm is None:
```

Comment:
This feels spooky... at a minimum we need to heavily document it. Is there another reasonable way to structure a "run infinitely" request?

Reply:
We could chop it? The only thing it buys you is the ability to drive in a circle for forever...not sure how useful that would be. They could always set the distance to some huge number to get the same end result.

Reply:
Yeah, I vote we chop it. There isn't really a reason to drive in a circle forever 😆